Installation issue: conduit: missing hdf5_hl.h

`conduit@master` is failing using `spack@develop` (31-July-2020, f3cb3a2eb8991cdfa131abcb429f13903911db1f) on:
* `CentOS 7` on `x86_64` and `ppc64le` using `gcc@8.1.0`
* `Ubuntu 18.04` on `x86_64` and `ppc64le` using `gcc@7.5.0`

I've attached the exact concretized spec file I'm using to do the install below.

All the dependencies are available in public build-cache: https://cache.e4s.io/e4s (gpg key here: [e4s.pub.txt](https://github.com/spack/spack/files/5014018/e4s.pub.txt))

1. Install dependencies from build-cache
```
$> spack gpg trust e4s.pub
$> spack mirror add e4s https://cache.e4s.io/e4s
$> spack install --cache-only --only dependencies -f conduit-tvtq22.yml
OK
```

2. Install `conduit` from source
```
$> spack install --no-cache --only package -f conduit-tvtq22.yml

==> Installing conduit
==> conduit: Executing phase: 'configure'
==> conduit: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

19 errors found in build log:
     117    -- Check size of long double - done
     118    -- Check size of void *
     119    -- Check size of void * - done
     120    -- RapidJSON headers found in: /tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/src/thir
            dparty_builtin/rapidjson/include
     121    -- Using RapidJSON Include: /tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/src/thirdpa
            rty_builtin/rapidjson/include
     122    -- Looking for HDF5 at: /opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj
  >> 123    /tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build/CMakeFiles/hdf5/cmake_hdf5_
            test.c:2:10: fatal error: hdf5_hl.h: No such file or directory
     124     #include <hdf5_hl.h>
     125              ^~~~~~~~~~~
     126    compilation terminated.
     127    -- HDF5 C compiler wrapper is unable to compile a minimal HDF5 program.
     128    -- Found HDF5: /opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj/lib/libhdf5.a (foun
            d version "1.8.21")
     129    -- HDF5_DIR_REAL=/opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj

     ...

     637    make[2]: Leaving directory `/tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build
            '
     638    [ 20%] Built target blt_gtest_smoke
     639    [ 21%] Linking CXX executable t_libyaml_smoke
     640    cd /tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build/tests/thirdparty && /opt
            /spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/cmake-3.18.0-krmbqpzangcr7iw76lydywq2rfyo5huq/bin/cmake -E cmake_link_script CMakeF
            iles/t_libyaml_smoke.dir/link.txt --verbose=1
     641    /opt/view/bin/g++ -O2 -g -DNDEBUG -rdynamic CMakeFiles/t_libyaml_smoke.dir/t_libyaml_smoke.cpp.o ../../thirdparty_builtin/libyaml-6
            90a781/CMakeFiles/conduit_libyaml.dir/src/api.c.o ../../thirdparty_builtin/libyaml-690a781/CMakeFiles/conduit_libyaml.dir/src/dumpe
            r.c.o ../../thirdparty_builtin/libyaml-690a781/CMakeFiles/conduit_libyaml.dir/src/emitter.c.o ../../thirdparty_builtin/libyaml-690a
            781/CMakeFiles/conduit_libyaml.dir/src/loader.c.o ../../thirdparty_builtin/libyaml-690a781/CMakeFiles/conduit_libyaml.dir/src/parse
            r.c.o ../../thirdparty_builtin/libyaml-690a781/CMakeFiles/conduit_libyaml.dir/src/reader.c.o ../../thirdparty_builtin/libyaml-690a7
            81/CMakeFiles/conduit_libyaml.dir/src/scanner.c.o ../../thirdparty_builtin/libyaml-690a781/CMakeFiles/conduit_libyaml.dir/src/write
            r.c.o -o t_libyaml_smoke  ../../lib/libgtest_main.a ../../lib/libgtest.a -lpthread -lpthread
     642    /opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj/lib/libhdf5.a(H5PL.o): In function
             `H5PL_term_interface':
  >> 643    H5PL.c:(.text+0x2d5): undefined reference to `dlclose'
     644    /opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj/lib/libhdf5.a(H5PL.o): In function
             `H5PL_load':
  >> 645    H5PL.c:(.text+0x7b5): undefined reference to `dlsym'
  >> 646    H5PL.c:(.text+0x9b1): undefined reference to `dlerror'
  >> 647    H5PL.c:(.text+0xaee): undefined reference to `dlopen'
  >> 648    H5PL.c:(.text+0xb09): undefined reference to `dlsym'
  >> 649    H5PL.c:(.text+0xbef): undefined reference to `dlclose'
  >> 650    H5PL.c:(.text+0xbfc): undefined reference to `dlclose'
  >> 651    H5PL.c:(.text+0xd32): undefined reference to `dlclose'
     652    /opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj/lib/libhdf5.a(H5Zdeflate.o): In fu
            nction `H5Z_filter_deflate':
  >> 653    H5Zdeflate.c:(.text+0x9c): undefined reference to `inflateInit_'
  >> 654    H5Zdeflate.c:(.text+0xb4): undefined reference to `inflate'
  >> 655    H5Zdeflate.c:(.text+0x209): undefined reference to `inflateEnd'
  >> 656    H5Zdeflate.c:(.text+0x278): undefined reference to `inflateEnd'
  >> 657    H5Zdeflate.c:(.text+0x2e3): undefined reference to `inflateEnd'
  >> 658    H5Zdeflate.c:(.text+0x38e): undefined reference to `compress2'
  >> 659    collect2: error: ld returned 1 exit status
  >> 660    make[2]: *** [tests/thirdparty/t_hdf5_smoke] Error 1
     661    make[2]: Leaving directory `/tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build
            '
  >> 662    make[1]: *** [tests/thirdparty/CMakeFiles/t_hdf5_smoke.dir/all] Error 2
     663    make[1]: *** Waiting for unfinished jobs....
     664    [ 21%] Linking CXX executable t_libb64_smoke
     665    cd /tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build/tests/thirdparty && /opt
            /spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/cmake-3.18.0-krmbqpzangcr7iw76lydywq2rfyo5huq/bin/cmake -E cmake_link_script CMakeF
            iles/t_libb64_smoke.dir/link.txt --verbose=1
     666    /opt/view/bin/g++ -O2 -g -DNDEBUG -rdynamic CMakeFiles/t_libb64_smoke.dir/t_libb64_smoke.cpp.o ../../thirdparty_builtin/libb64-1.2.
            1/CMakeFiles/conduit_b64.dir/src/cdecode.cpp.o ../../thirdparty_builtin/libb64-1.2.1/CMakeFiles/conduit_b64.dir/src/cencode.cpp.o -
            o t_libb64_smoke  ../../lib/libgtest_main.a ../../lib/libgtest.a -lpthread -lpthread
     667    [ 21%] Linking CXX executable t_rapidjson_smoke
     668    cd /tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build/tests/thirdparty && /opt
            /spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/cmake-3.18.0-krmbqpzangcr7iw76lydywq2rfyo5huq/bin/cmake -E cmake_link_script CMakeF
            iles/t_rapidjson_smoke.dir/link.txt --verbose=1

     ...

     678    [ 21%] Built target t_libb64_smoke
     679    [ 21%] Built target t_rapidjson_smoke
     680    /usr/bin/ranlib ../../lib/libconduit.a
     681    make[2]: Leaving directory `/tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build
            '
     682    [ 21%] Built target conduit
     683    make[1]: Leaving directory `/tmp/root/spack-stage/spack-stage-conduit-master-se4yzrhvntfzesfxaeyf7a5ccgygrjtc/spack-src/spack-build
            '
  >> 684    make: *** [all] Error 2
```

Full concretized spec: [conduit-tvtq22.yml.txt](https://github.com/spack/spack/files/5014016/conduit-tvtq22.yml.txt)
Full build log: [conduit-build-out.txt](https://github.com/spack/spack/files/5014013/conduit-build-out.txt)

Looks like a missing header file `hdf5_hl.h` ? Not present in the `hdf5` install tree as installed with Spack.

```
$> spack location -i hdf5
/opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj

$> find /opt/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/hdf5-1.8.21-tmcmgceuk7qoiu3mlenxkeg4ak63gfgj -name hdf5_hl.h
```

@ax3l @cyrush @bvanessen @scottwittenburg

--- a/var/spack/repos/builtin/packages/conduit/package.py
+++ b/var/spack/repos/builtin/packages/conduit/package.py
@@ -85,7 +85,7 @@ class Conduit(Package):
     # CMake
     #######################
     # cmake 3.8.2 or newer
-    depends_on("cmake@3.8.2:", type='build')
+    depends_on("cmake@3.8.2:3.17.9999", type='build')
 
     #######################
     # Python

